book/src/ephemeral-local-volumes.md

```diff
@@ -1,38 +1,49 @@
 
-The initial focus of Kubernetes CSI was Remote Persistent Volumes. However, the goal is for CSI to support all three types.
 
-This page documents how to create "Local Ephemeral Volumes" for Kubernetes using CSI.
+### Example of inline CSI pod spec
+A pod spec with an ephemeral inline CSI volume. Note that because the volume is expected to be ephemeral, the volumeHandle is not provided. Instead a CSI-generated ID will be submitted to the driver.
 
-# What is a Local Ephemeral Volumes?
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: some-pod
+spec:
+  containers:
+    ...
+  volumes:
+      - name: vol
+        csi:
+          driver: inline.storage.kubernetes.io
+          volumeAttributes:
+              foo: bar
+```
 
-A Local Ephemeral Volumes is a volume whose lifecycle is tied to the lifecycle of a single pod:
+## Implementing inline ephemeral support
 
-* The volume is "provisioned" (either empty or with some pre-populated data) when the pod is created.
-* The volume is deleted when the pod is terminated.
+Drivers must be modified (or implemented specifically) to support inline ephmeral workflows.  When Kubernetes encounters an inline CSI volume embedded in a pod spec, it treats that volume differently.  Mainly, the driver will only receive `NodePublish`, during the volume's mount phase, and `NodeUnpublish` when the pod is going away and the volume is unmounted.   To support inline, a driver must implement the followings:
 
-[Kubernetes Secret Volumes](https://kubernetes.io/docs/concepts/storage/volumes/#secret) are a good example (non-CSI) of a local ephemeral volumes.
+- Identity service
+- Node service
 
-# How to write a CSI Driver for Local Ephemeral Volumes
+### Feature gates
+To use inline volume, Kubernetes binaries must start with the `CSIInlineVolume` feature gate enabled:
+```
+--feature-gates=CSIInlineVolume=true
+```
 
-The following features make it easier to develop CSI Drivers that expose local ephemeral volumes:
-
-* [Pod Info on Mount](pod-info.md)
-  * This feature provides the CSI driver pod information at mount time. Many ephemeral volumes write some files at mount time. Often the data they write depends on the the pod they are operating on.
-* [Skip Attach](skip-attach.md)
-  * This instructs Kubernetes to skip any attach operation (`ControllerPublishVolume`) altogether. Local ephemeral volume drivers generally do not have or need a cluster control plane component.
-
-Features currently in development to improve Local Ephemeral Volume support:
-
-* Inline Volume Support
-  * Having to create a PV and a PVC for every ephemeral volume is onerous. Being able to specify a volume inside a pod definition (not currently possible for CSI drivers) will make that easier.
+### Example implementation
+- [CSI Hostpath driver](https://github.com/kubernetes-csi/csi-driver-host-path) - an example driver that supports either both PVC/PV and inline ephemeral volumes.
```

Comment:
"that supports either PVC/PV or inline ephemeral volumes"?

Reply:
Yeah, thats kind of unclear.

The driver supports both modes, but can only be launched in one mode at a time.

Reply:
Will revise wording.

Reply:
"either both"